### PR TITLE
Added gpt4all integration using langchain

### LINKED ISSUE #683: Support gpt4all integration via langchain
### 🚀 The feature

Current gpt4all integration is barebones and doesn't use langchain.
langchain provides a unified interface.
https://python.langchain.com/docs/integrations/providers/gpt4all

### Motivation, pitch

* When we move to using gpt4all via langchain, our users can use langsmith for observability.
* Standard interface for llm helps in better code maintenance

### PR DESCRIPTION
**Pull request by ShivamMenda**
## Description

Added gpt4 integration using langchain.

Fixes #683 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (does not change functionality, e.g. code style improvements, linting)
- [ ] Documentation update

## How Has This Been Tested?

Please delete options that are not relevant.

- [x] Unit Test
- [ ] Test Script (please provide)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

## Maintainer Checklist

- [ ] closes #683 
- [ ] Made sure Checks passed

### CODE REVIEW

**Comment by ShivamMenda** [comment]
@taranjeet any changes to be made wrt this PR let me know.

File: embedchain/llm/gpt4all.py [hunk 1]
```diff
@@ -19,13 +19,28 @@ def get_llm_model_answer(self, prompt):
...
+            max_tokens=BaseLlmConfig().max_tokens,
+            n_parts=8,
+            backend="llama",
+            seed=0,
+            f16_kv=False,
+            n_batch=8,
+            embedding=False,
+            logits_all=False,
+            vocab_only=False,
+            use_mlock=False,
```
Comment (deshraj) [inline embedchain/llm/gpt4all.py:42]: Do we really need to hard code these params? Can we skip these and let the GPT4All to pick the default ones?

Comment (ShivamMenda) [reply embedchain/llm/gpt4all.py:42]: @deshraj So sorry for the inactivity. will check if we can remove these hardcoded values.

**Comment by taranjeet** [comment]
thanks for you contribution here. we have added this support in main branch.

**Pull request closed without merging**